irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "torii/impl/command_service_transport_grpc.hpp"
+
+#include <iterator>
+
+#include "backend/protobuf/transaction.hpp"
+#include "backend/protobuf/transaction_responses/proto_tx_response.hpp"
+#include "builders/protobuf/transaction_responses/proto_transaction_status_builder.hpp"
+#include "builders/protobuf/transaction_sequence_builder.hpp"
+#include "common/timeout.hpp"
+#include "cryptography/default_hash_provider.hpp"
+#include "validators/default_validator.hpp"
+
+namespace torii {
+
+  CommandServiceTransportGrpc::CommandServiceTransportGrpc(
+      std::shared_ptr<CommandService> command_service,
+      std::shared_ptr<iroha::torii::StatusBus> status_bus,
+      std::chrono::milliseconds initial_timeout,
+      std::chrono::milliseconds nonfinal_timeout,
+      std::shared_ptr<shared_model::interface::TxStatusFactory> status_factory)
+      : command_service_(std::move(command_service)),
+        status_bus_(std::move(status_bus)),
+        initial_timeout_(initial_timeout),
+        nonfinal_timeout_(nonfinal_timeout),
+        status_factory_(std::move(status_factory)),
+        log_(logger::log("CommandServiceTransportGrpc")) {}
+
+  grpc::Status CommandServiceTransportGrpc::Torii(
+      grpc::ServerContext *context,
+      const iroha::protocol::Transaction *request,
+      google::protobuf::Empty *response) {
+    iroha::protocol::TxList single_tx_list;
+    *single_tx_list.add_transactions() = *request;
+    return ListTorii(context, &single_tx_list, response);
+  }
+
+  namespace {
+    /**
+     * Form an error message, which is to be shared between all transactions, if
+     * there are several of them, or individual message, if there's only one
+     * @param tx_hashes is non empty hash list to form error message from
+     * @param error of those tx(s)
+     * @return message
+     */
+    std::string formErrorMessage(
+        const std::vector<shared_model::crypto::Hash> &tx_hashes,
+        const std::string &error) {
+      if (tx_hashes.size() == 1) {
+        return (boost::format("Stateless invalid tx, error: %s, hash: %s")
+                % error % tx_hashes[0].hex())
+            .str();
+      }
+
+      std::string folded_hashes =
+          std::accumulate(tx_hashes.begin(),
+                          tx_hashes.end(),
+                          std::string(),
+                          [](auto &&acc, const auto &h) -> std::string {
+                            return acc + h.hex() + ", ";
+                          });
+
+      // remove leading ", "
```

Comment:
Please use the following approach to avoid any removals:
```C++
std::string s = std::accumulate(std::next(v.begin()), v.end(),
                                std::to_string(v[0]), // start with first element
                                [](std::string a, int b) {
                                    return a + '-' + std::to_string(b);
                                });
```